tests/conftest.py

```diff
@@ -133,3 +136,24 @@ def clean_and_restore_feedbacks(motion_controller):
     mc.configuration.set_velocity_feedback(vel, servo=alias)
     mc.configuration.set_position_feedback(pos, servo=alias)
     mc.configuration.set_auxiliar_feedback(aux, servo=alias)
+
+
+@pytest.hookimpl(tryfirst=True, hookwrapper=True)
+def pytest_runtest_makereport(item, call):
+    # execute all other hooks to obtain the report object
+    outcome = yield
+    rep = outcome.get_result()
+
+    # store test results for each phase of a call, which can be "setup", "call", "teardown"
+    item.stash.setdefault(test_report_key, {})[rep.when] = rep
+
+
+@pytest.fixture(scope="function", autouse=True)
+def load_configuration_if_test_fails(request, motion_controller, read_config):
+    mc, alias = motion_controller
+    yield
+
+    report = request.node.stash[test_report_key]
```

Comment:
Could you let me know why you use request.node.stash instead item.stash? I don't know the difference but in documentation the second case is used.
https://docs.pytest.org/en/7.1.x/how-to/writing_hook_functions.html#storing-data-on-items-across-hook-functions

Reply:
I'm following the [pytest example code](https://docs.pytest.org/en/latest/example/simple.html#making-test-result-information-available-in-fixtures). I'm not sure about the difference between those two.